src/server/svr_func.c

```diff
@@ -5277,6 +5310,48 @@ is_vnode_prov_done(char * vnode)
 	set_task(WORK_Immed, 0, do_provisioning, (void *) NULL);
 }
 
+/**
+ * @brief
+ *	Call check_and_run_jobs(prov_vnode_info) from a timed task to ensure that
+ *	hooks are synced with the node before job get start
+ *
+ * @param[in,out]       ptask - work task structure contains prov_vnode_info
+ */
+
+static void 
+is_sync_hooks_done(struct work_task *ptask)
+{
+	struct  pbsnode         *pnode = NULL;
+	struct prov_vnode_info * prov_vnode_info;
+
+	assert(ptask->wt_parm1 != NULL);
+
+	prov_vnode_info = (struct prov_vnode_info *) ptask->wt_parm1;
+	pnode = (struct pbsnode *) find_nodebyname(prov_vnode_info->pvnfo_vnode);
+
+	DBPRT(("%s: calling [check_and_run_jobs] from is_sync_hooks_done\n", __func__))
+
+	if (check_and_run_jobs(prov_vnode_info) == -5)
+		/**
+		 * No need to remove prov record 
+		 * as time task has been created
+		 */
+		return;
```

Comment:
We need to have a limit on number of attempts to check the hook_sync. And re-queue the job if sync is failing